### PR TITLE
Simulate the case when controller remove a controlee.

### PR DESCRIPTION
**Pull request by bobwzy**
When a controller removes a controlee, the controlee should stop ranging
automatically instead waiting for a timeout. So added a new PicaCommand to
achieve this.

Also remove the AppConfig that only applicable for controller from AppConfig
matching.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -590,6 +614,27 @@ impl Pica {
         }
     }
 
+    async fn stop_controlee_ranging(
```
Comment (hchataing) [inline src/lib.rs:617]: Same, please add some context here describing what this function is emulating.

File: src/lib.rs [hunk 2]
```diff
@@ -160,6 +160,8 @@ pub enum PicaCommand {
     Disconnect(usize),
     // Execute ranging command for selected device and session.
     Ranging(usize, u32),
+    // Stop ranging session with certain session id and app config.
```
Comment (hchataing) [inline src/lib.rs:163]: `Send an in-band request to stop ranging to a peer controlee identified by address`
I am not sure why the session id and app config would be required here, can you expand ?
You would need to controller address though.

File: src/session.rs [hunk 3]
```diff
@@ -834,17 +837,30 @@ impl Session {
...
                     } else {
-                        dst_addresses
-                            .retain(|value| *value != MacAddress::Short(controlee.short_address));
+                        dst_addresses.retain(|value| *value != MacAddress::Short(address));
+
+                        tokio::spawn(async move {
+                            pica_tx
+                                .send(PicaCommand::StopRanging(
+                                    MacAddress::Short(address),
+                                    session_id,
```
Comment (hchataing) [inline src/session.rs:855]: The session ID should be the session ID on the controlee device, not the local session ID.
Same for the App config.

**Review by hchataing — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (bobwzy) [reply src/lib.rs:163]: I supposed we need the session id to get the corresponding session on Controlee side. But it is not mentioned in the spec, so  seems the session_id doesn't have to be identical on both sides?

Then maybe use controller address is enough.

Comment (hchataing) [reply src/lib.rs:163]: I assume the session id is local to the uwbs instance, and not shared over the air, but I may be wrong as I have not read into the exact exchange format.

Comment (bobwzy) [reply src/lib.rs:163]: There's the definition in the CSML spec:
UWB Session ID: A 32bit unique integer identifying a UWB Session, shared between Controller and Controlee.
https://groups.firaconsortium.org/wg/Technical/document/2415

So I think using session_id to find the session is sufficient.

Comment (hchataing) [reply src/lib.rs:163]: Ok, but then I am not certain we properly implement session ID management in Pica.
From what I am reading the controller session ID is used as identifier for the session.

Comment (hchataing) [reply src/lib.rs:163]: I am not certain what happens to the session ID on the controlee side

File: src/session.rs (hunk 3 above)
Comment (bobwzy) [reply src/session.rs]: Removed the app config. As discussed above I think we can use session id to get the session we want.

**Pull request merged**